### PR TITLE
GBI-643

### PR DESCRIPTION
**Pull request by joaoiovlabs**
#Title
Add an awaiter for the transaction to finish and display a message of successfulness (use the toast)

#Description
It was added spinner when user swap token.
It was added error message builder.
it was added success message notification.
It was changed global states and actions.

### CODE REVIEW

File: src/components/shared/WaitSpinner.vue [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<template>
+  <div
+    v-if="show"
+    id="wait"
+    class="mt-3 align-center text-center"
+    style="max-width: 350px; line-height: 16px"
+  >
+    <!-- Spinner <div class="spinner-border mb-4" style="height:70px; width:70px"></div> -->
```
Comment (coolwather) [inline src/components/shared/WaitSpinner.vue:8]: Remove this comment to let the code clean.

File: src/components/swap/SwapBox.vue [hunk 2]
```diff
@@ -319,17 +372,55 @@ export default defineComponent({
               s: signedData.s,
               estimatedGasFee,
               sideTokenBtcContract: this.swapFrom.address,
+              deadline: deadline.toString(),
```
Comment (aguschirico) [inline src/components/swap/SwapBox.vue:375]: why are you adding this here?

File: src/utils/transactions.ts [hunk 3]
```diff
@@ -18,15 +18,10 @@ export function transactionCallback({ resolve, reject, web3, explorer }) {
...
+        const url = `${process.env.VUE_APP_EXPLORER_ADDRESS}/tx/${txHash}`;
         return reject(
           new Error(
-            `Transaction status failed ${textExplorerLink}
-                <p class="d-none">
-                  <small>Transaction Info</small>
-                  <pre class="overflow-auto d-none" style="max-height: 200px">${stringReceipt}</pre>
-                </p>
-              `
+            `<span>Error: Transaction Info <a class="error-message" href="${url}">${url}</a></span>`
```
Comment (aguschirico) [inline src/utils/transactions.ts:24]: this should open in a new tab, not in the same tab you are in. Please add `target="_blank"` to the anchor.

File: src/components/shared/Notification.vue [hunk 4]
```diff
@@ -55,7 +67,7 @@ export default {
     };
   },
   watch: {
-    error() {
+    message() {
```
Comment (aguschirico) [inline src/components/shared/Notification.vue:70]: remember we are planning to add at least one test to every PR.

**Review by aguschirico — CHANGES_REQUESTED** [review]
Great work! just a few comments.

**Review by aguschirico — APPROVED** [review]

**Pull request merged**